app/repository/discount.py

```diff
@@ -0,0 +1,91 @@
+"""Discount repository module."""
+
+from collections.abc import Sequence
+from datetime import UTC, datetime
+from typing import Any
+from uuid import UUID
+
+import sqlalchemy as sa
+from sqlalchemy import null, or_
+
+from app.db.model import Discount
+from app.repository.base import BaseRepository
+
+
+class DiscountRepository(BaseRepository):
+    """Repository for managing discount records in the database."""
+
+    async def get_current_vlab_discount(self, vlab_id: UUID) -> Discount | None:
+        """Retrieve the currently active discount for a specific virtual lab.
+
+        Finds the most recent discount that is currently valid (started in the past and
+        either has no end date or ends in the future).
+
+        Args:
+            vlab_id: UUID of the virtual lab to query discounts for
+
+        Returns:
+            The most recent active Discount object or None if no active discount exists
+        """
+        now = datetime.now(tz=UTC)
```

Comment:
Minor, there is `utils.utcnow()` that would be easier to mock or change if needed in the future.

Reply:
Good point, fixed.